modules/core/src/test/scala/org/tessellation/infrastructure/rewards/RewardsSuite.scala

```diff
@@ -2,141 +2,109 @@ package org.tessellation.infrastructure.rewards
 
 import cats.data.NonEmptySet
 import cats.effect.std.Random
+import cats.effect.unsafe.implicits.global
 import cats.effect.{IO, Resource}
 import cats.implicits.catsSyntaxApplicativeId
 import cats.syntax.contravariantSemigroupal._
-
-import scala.collection.immutable.SortedSet
+import cats.syntax.list._
 
 import org.tessellation.config.types._
 import org.tessellation.dag.dagSharedKryoRegistrar
 import org.tessellation.dag.snapshot.epoch.EpochProgress
 import org.tessellation.ext.kryo._
+import org.tessellation.keytool.KeyPairGenerator
 import org.tessellation.kryo.KryoSerializer
 import org.tessellation.schema.ID.Id
-import org.tessellation.schema.balance.Amount
+import org.tessellation.schema.generators.chooseNumRefined
 import org.tessellation.sdk.sdkKryoRegistrar
 import org.tessellation.security.SecurityProvider
-import org.tessellation.security.hash.Hash
-import org.tessellation.security.hex.Hex
+import org.tessellation.security.key.ops.PublicKeyOps
 
 import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.NonNegLong
+import org.scalacheck.Gen
 import weaver.MutableIOSuite
 import weaver.scalacheck.Checkers
 
 object RewardsSuite extends MutableIOSuite with Checkers {
-  type Res = (KryoSerializer[IO], SecurityProvider[IO], Random[IO])
+  type GenIdFn = () => Id
+  type Res = (KryoSerializer[IO], SecurityProvider[IO], Random[IO], GenIdFn)
 
   override def sharedResource: Resource[IO, Res] =
     (
       KryoSerializer.forAsync[IO](dagSharedKryoRegistrar.union(sdkKryoRegistrar)),
       SecurityProvider.forAsync[IO],
       Resource.liftK(Random.scalaUtilRandom[IO])
-    ).mapN((_, _, _))
-
-  implicit val order: Ordering[Id] = (x: Id, y: Id) => 1
+    ).mapN {
+      case (ks, sp, r) =>
+        implicit val _sp: SecurityProvider[IO] = sp
+        (ks, sp, r, () => KeyPairGenerator.makeKeyPair.map(_.getPublic.toId).unsafeRunSync())
```

Comment:
I couldn't reuse `idGen` because... it generates incorrect `Id`s in terms of `Id => Address` conversion.